fedimint-server/src/consensus/engine.rs

```diff
@@ -391,12 +389,14 @@ impl ConsensusEngine {
         // signature or a signed session outcome arrives from our peers
         while signatures.len() < self.keychain.threshold() {
             tokio::select! {
-                unit_data = unit_data_receiver.recv() => {
-                    if let (UnitData::Signature(signature), peer) = unit_data? {
-                        if self.keychain.verify(&header, &signature, to_node_index(peer)){
-                            signatures.insert(peer, signature);
+                ordered_unit = ordered_unit_receiver.recv() => {
+                    let ordered_unit = ordered_unit?;
+
+                    if let Some(UnitData::Signature(signature)) = ordered_unit.data {
```

Comment:
nit: could use [`let-else` syntax](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html) for early return to reduce indentation, e.g.

```rust
let Some(UnitData::Signature(signature)) = ordered_unit.data else { continue; };
```